File: leetcode-04/solutions/python3/0974.py
class Solution:
    def subarraysDivByK(self, A: List[int], K: int) -> int:
        ans = 0
        sum = 0
        map = [1] + [0] * (K - 1)

        for a in A:
            sum = (sum + a) % K
            ans += map[sum]
            map[sum] += 1

        return ans

File: leetcode-04/solutions/cpp/0974.cpp
class Solution {
 public:
  int subarraysDivByK(vector<int>& A, int K) {
    int ans = 0;
    int sum = 0;
    vector<int> map(K);
    map[0] = 1;

    for (int a : A) {
      sum = (sum + a) % K;
      if (sum < 0) sum += K;
      ans += map[sum];
      map[sum]++;
    }

    return ans;
  }
};